source/common/network/io_socket_handle_impl.cc

```diff
@@ -605,7 +604,7 @@ Address::InstanceConstSharedPtr IoSocketHandleImpl::localAddress() {
     throw EnvoyException(fmt::format("getsockname failed for '{}': ({}) {}", fd_, result.errno_,
                                      errorDetails(result.errno_)));
   }
-  return Address::addressFromSockAddr(ss, ss_len, socket_v6only_);
+  return *Address::addressFromSockAddr(ss, ss_len, socket_v6only_);
```

Comment:
`addressFromSockAddrOrThrow` or `addressFromSockAddrOrDie`

Look for all of the call-sites where you are applying `*` without checking to see whether the object is valid. Based on the doc for absl::Status that feels like undefined behavior, and we should either explicitly all the `OrThrow` variant or the `OrDie` variant.

Reply:
probably you want `ASSERT(Thread::MainThread::isMainThread())` at the top of this function also.

Reply:
Makes sense.